packages/fluentui/react-northstar/src/components/Dropdown/Dropdown.tsx

```diff
@@ -826,31 +817,154 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
   };
 
   handleStateChange = (changes: StateChangeOptions<ShorthandValue<DropdownItemProps>>) => {
-    if (changes.isOpen !== undefined && changes.isOpen !== this.state.open) {
-      const newState = { open: changes.isOpen, highlightedIndex: this.state.highlightedIndex };
-
-      if (changes.isOpen) {
-        const highlightedIndexOnArrowKeyOpen = this.getHighlightedIndexOnArrowKeyOpen(changes);
-        if (_.isNumber(highlightedIndexOnArrowKeyOpen)) {
-          newState.highlightedIndex = highlightedIndexOnArrowKeyOpen;
+    const { search, multiple, highlightFirstItemOnOpen, items, getA11ySelectionMessage } = this.props;
+    const { value, open } = this.state;
+    const {
+      type,
+      inputValue: newSearchQuery,
+      selectedItem: newValue,
+      isOpen: newOpen,
+      highlightedIndex: newHighlightedIndex,
+    } = changes;
+    const newState = {} as DropdownState;
+    const handlers = [];
+
+    switch (type) {
+      case Downshift.stateChangeTypes.changeInput: {
+        const shouldValueChange = newSearchQuery === '' && !multiple && !!value.length;
+        newState.searchQuery = newSearchQuery;
+        newState.highlightedIndex = highlightFirstItemOnOpen ? 0 : null;
+
+        if (shouldValueChange) {
+          newState.value = [];
         }
-        if (!this.props.search) {
-          this.listRef.current.focus();
+
+        if (open) {
+          // we clear value when in single selection user cleared the query.
+          const shouldMenuClose = newSearchQuery === '' || newValue !== undefined;
+
+          if (shouldMenuClose) {
+            newState.open = false;
+          }
+        } else {
+          newState.open = true;
         }
-      } else {
-        newState.highlightedIndex = null;
+
+        break;
       }
+      case Downshift.stateChangeTypes.keyDownEnter:
+      case Downshift.stateChangeTypes.clickItem:
+        const shouldAddHighlightedIndex = !multiple && !!items;
+
+        newState.searchQuery = this.getSelectedItemAsString(newValue);
+        newState.value = multiple ? [...value, newValue] : [newValue];
+        newState.open = false;
+
+        if (shouldAddHighlightedIndex) {
+          newState.highlightedIndex = items.indexOf(newValue);
+        }
+
+        if (getA11ySelectionMessage && getA11ySelectionMessage.onAdd) {
+          this.setA11ySelectionMessage(getA11ySelectionMessage.onAdd(newValue));
+        }
+
+        if (multiple) {
+          setTimeout(() => (this.selectedItemsRef.current.scrollTop = this.selectedItemsRef.current.scrollHeight), 0);
+        }
+
+        this.tryFocusTriggerButton();
```

Comment:
Can be this and `setTimeout()` be moved to `componentDidUpdate()`?

Reply:
Since it's not straightforward to do and not included in scope, I'd rather leave it like this for now.

Reply:
Agree, let's leave it here for now, but at least consider this for future in other components, too